### PR TITLE
[mergebot] Make merge on green default behavior

### PR DESCRIPTION
**Pull request by zengk95**
This makes the merge on green behavior default as we tried to do it earlier, but forgot to pass in the force arguments.

This also adds a few tests so that we don't make that mistake again (although people can still forget to pass in the arguments through shell).

### CODE REVIEW

**Review by janeyx99 — APPROVED** [review]
This looks okay but will have massive conflicts with https://github.com/pytorch/pytorch/pull/79196 cc @malfet

**Review by seemethere — APPROVED** [review]

**Review by janeyx99 — APPROVED** [review]

File: .github/scripts/test_trymerge.py [hunk 1]
```diff
@@ -30,7 +30,6 @@ def save_mocked_queries(obj: Any) -> None:
         with open(gql_db_fname, encoding="utf-8", mode="w") as f:
             json.dump(obj, f, indent=2)
             f.write("\n")
-
```
Comment (malfet) [inline .github/scripts/test_trymerge.py:33]: Why this change is needed here?

File: .github/scripts/trymerge.py [hunk 2]
```diff
@@ -972,16 +974,15 @@ def handle_exception(e: Exception, msg: str = "Merge failed") -> None:
...
-        try:
-            pr.merge_into(repo, dry_run=args.dry_run, force=args.force, comment_id=args.comment_id)
-        except Exception as e:
-            handle_exception(e)
+    try:
+        merge(args.pr_num, repo,
+              dry_run=args.dry_run,
+              force=args.force,
+              comment_id=args.comment_id,
+              mandatory_only=args.mandatory_only)
```
Comment (malfet) [inline .github/scripts/trymerge.py:982]: There are no `mandatory_only` options in the args anymore, so this would always be false, isn't it? If you are deleting it from the options, perhaps you shall either delete it or keep the option (even if it can not be invoked via the bot)

**Review by malfet — APPROVED** [review]
I would appreciate if there would still be an option to just try to merge without any busy waiting

**Review by janeyx99 — APPROVED** [review]
